### PR TITLE
Feat/autorollover

### PR DESCRIPTION
**Pull request by avniculae**

### CODE REVIEW

File: src/entities/mellow/mellowLpRouter.ts [hunk 1]
```diff
@@ -80,6 +80,8 @@ class MellowLpRouter {
 
   public isRegisteredForAutoRollover = false;
 
+  private canManageVaultPositions?: boolean[];
```
Comment (ssbarbee) [inline src/entities/mellow/mellowLpRouter.ts:83]: does it have to be nullable? you can just do
```suggestion
  private canManageVaultPositions: boolean[] = [];
```

File: src/entities/mellow/mellowLpRouter.ts [hunk 2]
```diff
@@ -840,6 +858,18 @@ class MellowLpRouter {
   public get autoRolloverRegistrationGasFeeUSD() {
     return this.autoRolloverRegistrationGasUnits * this.gasUnitPriceUSD;
   }
+
+  public canManageVaultPosition = (vaultIndex: number): boolean => {
+    if (this.canManageVaultPositions === undefined) {
+      return false;
+    }
+
+    if (!(vaultIndex < this.vaultsCount)) {
```
Comment (ssbarbee) [inline src/entities/mellow/mellowLpRouter.ts:867]: why not write a more readable code?

```suggestion
    if (vaultIndex >= this.vaultsCount) {
```

Comment (ssbarbee) [inline src/entities/mellow/mellowLpRouter.ts:867]: what happens if integrator calls with negative index?

File: tests/mocha/mellowLpRouter.test.ts [hunk 3]
```diff
@@ -1062,7 +1046,121 @@ describe('Mellow Router Test Suite', () => {
...
+          deprecated: false,
+          vaults: [],
+          underlyingPools: [],
+        },
+      });
+      await ethMellowLpRouter.vaultInit();
+      await ethMellowLpRouter.userInit(userWallet);
+
+      for (let vaultIndex = 0; vaultIndex < ethMellowLpRouter.vaultsCount; vaultIndex += 1) {
+        expect(ethMellowLpRouter.canManageVaultPosition(vaultIndex)).to.be.eq(true);
```
Comment (ssbarbee) [inline tests/mocha/mellowLpRouter.test.ts:1083]: missing test cases with negative indexes, with indexes larger than vaults count

File: src/entities/mellow/mellowLpRouter.ts (hunk 1 above)
Comment (avniculae) [reply src/entities/mellow/mellowLpRouter.ts:83]: I think it helps to keep its length consistent with the length of the ERC20RootVaultAddresses vector.

Comment (ssbarbee) [reply src/entities/mellow/mellowLpRouter.ts:83]: then does it make sense to keep this as property in the class if you can compute it using another property?

File: tests/mocha/mellowLpRouter.test.ts (hunk 3 above)
Comment (avniculae) [reply tests/mocha/mellowLpRouter.test.ts:1083]: addressed

File: src/entities/mellow/mellowLpRouter.ts (hunk 2 above)
Comment (avniculae) [reply src/entities/mellow/mellowLpRouter.ts]: addressed

**Review by ssbarbee — APPROVED** [review]
